client/app/bundles/course/group/pages/GroupShow/GroupManager/GroupUserManager.jsx

```diff
@@ -118,22 +130,65 @@ const getAvailableUsers = (
   groups,
   group,
   hideInGroup,
+  hidePhantomStudent,
   availableSearch,
 ) => {
+  let groupMemberIds = new Set()
+  // const altGroup = new Set(groups.flatMap((g) => g.members).filter((m) => !m.isPhantom));
   if (hideInGroup) {
-    const allGroupMemberIds = new Set(
-      groups.flatMap((g) => g.members.map((m) => m.id)),
-    );
-    return filterByName(
-      availableSearch,
-      courseUsers.filter((cu) => !allGroupMemberIds.has(cu.id)),
-    );
+    groupMemberIds = new Set(groups.flatMap((g) => g.members.map((m) => m.id)));
+  } else {
+    groupMemberIds = new Set(group.members.map((m) => m.id))
   }
-  const groupMemberIds = new Set(group.members.map((m) => m.id));
-  return filterByName(
-    availableSearch,
+
+  const filteredGroup = filterByName(availableSearch,
     courseUsers.filter((cu) => !groupMemberIds.has(cu.id)),
   );
+
+  if (hidePhantomStudent) {
+    return filteredGroup.filter((m) => !m.isPhantom);
+  } 
+  return filteredGroup
+};
+
+const getSelectedUsers = (
+  members,
+  selectedSearch,
+  hidePhantomStudent,
+) => {
+  let groupMembers = new Set()
+  if (hidePhantomStudent) {
+    groupMembers = new Set(members.filter((m) => !m.isPhantom))
+  } else {
+    groupMembers = new Set(members)
+  }
+
+  return filterByName(
+    selectedSearch,
+    [...groupMembers],
+  )
+};
+
+const getAvailableUserInOtherGroups = (
+  courseUsers,
+  groups,
+  group,
+) => {
+  const otherGroups = groups.filter((x) => x !== group)
+  const mapStudentToGroups = {}
```

Comment:
It's better to store the value of the object as an array containing the group names. When needed later, we can just use `join`.

Reply:
actually that mapStudentToGroups does not only contain info of groups attended by one student. It has the informations of each student joining which groups. i don't think array is the good data structure to be used here

Reply:
resolved